### PR TITLE
feat: User Login

### PR DESCRIPTION
**Pull request by bluntbrain**

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,12 +1,17 @@
 import React from 'react';
+import {StatusBar} from 'react-native';
 import {NavigationContainer} from '@react-navigation/native';
+import FlashMessage from 'react-native-flash-message';
 
 import AppStack from './src/navigation/AppStack';
+import colors from './src/theme/colors';
 
 function App(): JSX.Element {
```
Comment (atulpatel-03) [inline App.tsx:9]: All new components will be functional components

File: src/components/common/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React, {useState} from 'react';
+import {
+  TouchableOpacity,
+  Text,
+  ActivityIndicator,
+  StyleSheet,
+} from 'react-native';
+import colors from '../../theme/colors';
```
Comment (atulpatel-03) [inline src/components/common/Button.tsx:8]: Library imports on top, a space and then rest imports

File: src/components/common/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    backgroundColor: colors.gray,
+  },
+  buttonText: {
+    color: colors.black,
+    fontSize: 16,
+    fontWeight: '500',
+  },
+});
+
+const Button = ({title, onPress, disabled = false}) => {
```
Comment (atulpatel-03) [inline src/components/common/Button.tsx:27]: For Props, Please define its TS Type

File: src/components/common/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  const [is_loading, set_is_loading] = useState(false);
```
Comment (atulpatel-03) [inline src/components/common/Button.tsx:28]: For ever useState, define its TS type ...Like useState<boolean>

File: src/screens/UserOnboardingAndAuth/Login.tsx [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+
+import {useOnboardingAndAuth} from './useOnboardingAndAuth';
+import PhoneNumberInput from './components/PhoneNumberInput';
+import OTPVerification from './components/OTPVerification';
+import {AuthContext} from './context';
+
+const LoginComponent = () => {
+  const {confirm} = useContext(AuthContext);
+
+  console.log('confirm', confirm);
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/Login.tsx]: remove console.log  while pushing the code

File: src/screens/UserOnboardingAndAuth/components/OTPVerification.tsx [hunk 6]
```diff
@@ -0,0 +1,86 @@
+import {StyleSheet, Text, TouchableOpacity, View, Image} from 'react-native';
+import React, {useContext} from 'react';
+import OTPInputView from '@twotalltotems/react-native-otp-input';
+import {AuthContext} from '../context';
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/components/OTPVerification.tsx:4]: Library imports on top, a space and then rest imports

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
+import {useState} from 'react';
+import auth from '@react-native-firebase/auth';
+
+import {show_flash_message} from '../../utils/helper';
+
+export const useOnboardingAndAuth = () => {
+  const [phone_number, set_phone_number] = useState('');
+  const [confirm, set_confirm] = useState(null);
+  const [code, set_code] = useState('');
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:9]: Define TS type for all the states

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+
+    const valid_phone_number = '+91' + sanitized_number;
+
+    const confirmation = await auth().signInWithPhoneNumber(valid_phone_number);
+    set_confirm(confirmation);
+  }
+  async function confirm_code() {
+    try {
+      await confirm.confirm(code);
+      console.log('Confirmed!');
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:32]: Remove console.log

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+    } catch (error) {
+      console.log('Invalid code.');
```
Comment (atulpatel-03) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:34]: Remove here also

File: src/components/common/Button.tsx [hunk 10]
```diff
@@ -24,8 +25,14 @@ const styles = StyleSheet.create({
   },
 });
 
-const Button = ({title, onPress, disabled = false}) => {
-  const [is_loading, set_is_loading] = useState(false);
+interface ButtonProps {
```
Comment (atulpatel-03) [inline src/components/common/Button.tsx:28]: Only Give Props as name

Comment (bluntbrain) [reply src/components/common/Button.tsx:28]: discussed

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+      );
+      return;
+    }
+
+    const valid_phone_number = '+91' + sanitized_number;
+
+    const confirmation = await auth().signInWithPhoneNumber(valid_phone_number);
+    set_confirm(confirmation);
+  }
+  async function confirm_code() {
```
Comment (shreyanshgoel9) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:29]: arrow functions only

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+        'danger',
+      );
+      return;
+    }
+
+    const valid_phone_number = '+91' + sanitized_number;
+
+    const confirmation = await auth().signInWithPhoneNumber(valid_phone_number);
+    set_confirm(confirmation);
+  }
```
Comment (shreyanshgoel9) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:28]: give space after functions

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx [hunk 13]
```diff
@@ -0,0 +1,51 @@
+import {useState} from 'react';
+import auth from '@react-native-firebase/auth';
+
+import {show_flash_message} from '../../utils/helper';
+
+export const useOnboardingAndAuth = () => {
```
Comment (shreyanshgoel9) [inline src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:6]: use export default when exporting the main function

File: src/screens/UserOnboardingAndAuth/components/PhoneNumberInput.tsx [hunk 14]
```diff
@@ -0,0 +1,96 @@
...
+    padding: 20,
+  },
+});
+
+const PhoneNumberInput = () => {
+  const {phone_number, set_phone_number, sign_in_with_phone_number} =
+    useContext(AuthContext);
+
+  return (
+    <View style={styles.container}>
```
Comment (shreyanshgoel9) [inline src/screens/UserOnboardingAndAuth/components/PhoneNumberInput.tsx:65]: seems like eslint is not setup, please see to it that its setup

**Review by shreyanshgoel9 — CHANGES_REQUESTED** [review]

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx (hunk 11 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:29]: resolved

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx (hunk 12 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:28]: resolved

File: src/screens/UserOnboardingAndAuth/components/PhoneNumberInput.tsx (hunk 14 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/components/PhoneNumberInput.tsx:65]: resolved

File: src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx (hunk 13 above)
Comment (bluntbrain) [reply src/screens/UserOnboardingAndAuth/useOnboardingAndAuth.tsx:6]: resolved

**Pull request merged**